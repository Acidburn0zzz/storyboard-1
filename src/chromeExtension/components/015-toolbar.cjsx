_                 = require '../../vendor/lodash'
React             = require 'react'
ReactRedux        = require 'react-redux'
{ Icon }          = require 'giu'
Login             = require './010-login'
Settings          = require './016-settings'
actions           = require '../actions/actions'

mapStateToProps = (state) -> 
  settings:       state.settings
  wsState:        state.cx.wsState
mapDispatchToProps = (dispatch) ->
  expandAllStories:   -> dispatch actions.expandAllStories()
  collapseAllStories: -> dispatch actions.collapseAllStories()
  clearLogs:          -> dispatch actions.clearLogs()
  quickFind: (txt) -> dispatch actions.quickFind txt

Toolbar = React.createClass
  displayName: 'Toolbar'

  #-----------------------------------------------------
  propTypes:
    # From Redux.connect
    settings:             React.PropTypes.object.isRequired
    wsState:              React.PropTypes.string.isRequired
    expandAllStories:     React.PropTypes.func.isRequired
    collapseAllStories:   React.PropTypes.func.isRequired
  getInitialState: ->
    fSettingsShown:       false


  #-----------------------------------------------------
  render: -> 
    <div>
      {@renderSettings()}
      <div style={_style.outer}>
        <div style={_style.left}>
          <Icon 
            icon="cog" 
            size="lg"
            title="Show settings..."
            onClick={@toggleSettings}
            style={_style.icon}
          />
          <Icon 
            icon="chevron-circle-down" 
            size="lg" 
            title="Expand all stories"
            onClick={@props.expandAllStories}
            style={_style.icon}
          />
          <Icon 
            icon="chevron-circle-right" 
            size="lg" 
            title="Collapse all stories"
            onClick={@props.collapseAllStories}
            style={_style.icon}
          />
          <Icon 
            icon="remove"
            size="lg" 
            title="Clear logs"
            onClick={@props.clearLogs}
            style={_style.icon}
          />
          {' '}
          <input
            id="quickFind"
            type="search"
            results=0
            placeholder="Quick find..."
            onChange={@onChangeQuickFind}
          />
          {@renderWsStatus()}
        </div>
        <div style={_style.spacer}/>
        <Login/>
      </div>
      <div style={_style.placeholder}/>
    </div>

  renderSettings: ->
    return if not @state.fSettingsShown
    <Settings onClose={@toggleSettings}/>

  renderWsStatus: ->
    fConnected = @props.wsState is 'CONNECTED'
    if fConnected
      icon = 'chain'
      title = 'Connection with the server is UP'
    else
      icon = 'chain-broken'
      title = 'Connection with the server is DOWN'
    <Icon 
      icon={icon}
      size="lg" 
      title={title}
      style={_style.wsStatus fConnected}
    />

  #-----------------------------------------------------
  toggleSettings: -> @setState {fSettingsShown: not @state.fSettingsShown}
  onChangeQuickFind: (ev) -> @props.quickFind ev.target.value

#-----------------------------------------------------
_style = 
  outer:
    position: 'fixed'
    top: 0
    left: 0
    height: 30
    width: '100%'
    backgroundColor: 'white'
    borderBottom: '1px solid #ccc'
    display: 'flex'
    flexDirection: 'row'
    whiteSpace: 'nowrap'
    zIndex: 10
  icon:
    cursor: 'pointer'
    color: '#666'
    marginRight: 10
  wsStatus: (fConnected) ->
    marginRight: 5
    marginLeft: 10
    color: if fConnected then 'green' else 'red'
    cursor: 'default'
  placeholder:
    height: 30
  left:
    padding: "4px 4px 4px 8px"
  right:
    padding: "4px 8px 4px 4px"
  spacer:
    flex: '1 1 0px'

#-----------------------------------------------------
connect = ReactRedux.connect mapStateToProps, mapDispatchToProps
module.exports = connect Toolbar
