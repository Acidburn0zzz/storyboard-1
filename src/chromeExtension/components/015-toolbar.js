import React from 'react';
import * as ReactRedux from 'react-redux';
import { Icon, isDark } from 'giu';
import Login from './010-login';
import Settings from './016-settings';
import actions from '../actions/actions';  // eslint-disable-line import/no-unresolved, import/extensions

const mapStateToProps = (state) => ({
  wsState: state.cx.wsState,
});
const mapDispatchToProps = {
  expandAllStories: actions.expandAllStories,
  collapseAllStories: actions.collapseAllStories,
  clearLogs: actions.clearLogs,
  quickFind: actions.quickFind,
};

class Toolbar extends React.PureComponent {
  static propTypes = {
    colors: React.PropTypes.object.isRequired,
    // From Redux.connect
    wsState: React.PropTypes.string.isRequired,
    expandAllStories: React.PropTypes.func.isRequired,
    collapseAllStories: React.PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = {
      fSettingsShown: false,
    };
  }

  // -----------------------------------------------------
  render() {
    const { colors } = this.props;
    return (
      <div>
        {this.renderSettings()}
        <div style={style.outer(colors)}>
          <div style={style.left}>
            <Icon
              id="sbBtnShowSettings"
              icon="cog"
              size="lg"
              title="Show settings..."
              onClick={this.toggleSettings}
              style={style.icon(colors)}
            />
            <Icon
              icon="chevron-circle-down"
              size="lg"
              title="Expand all stories"
              onClick={this.props.expandAllStories}
              style={style.icon(colors)}
            />
            <Icon
              icon="chevron-circle-right"
              size="lg"
              title="Collapse all stories"
              onClick={this.props.collapseAllStories}
              style={style.icon(colors)}
            />
            <Icon
              icon="remove"
              size="lg"
              title="Clear logs"
              onClick={this.props.clearLogs}
              style={style.icon(colors)}
            />
            {' '}
            <input
              id="quickFind"
              type="search"
              results={0}
              placeholder="Quick find..."
              onChange={this.onChangeQuickFind}
              style={style.quickFind(colors)}
            />
            {this.renderWsStatus()}
          </div>
          <div style={style.spacer} />
          <Login colors={colors} />
        </div>
        <div style={style.placeholder} />
      </div>
    );
  }

  renderSettings() {
    if (!this.state.fSettingsShown) return null;
    return (
      <Settings
        onClose={this.toggleSettings}
        colors={this.props.colors}
      />
    );
  }

  renderWsStatus() {
    const fConnected = this.props.wsState === 'CONNECTED';
    const icon = fConnected ? 'chain' : 'chain-broken';
    const title = fConnected
      ? 'Connection with the server is UP'
      : 'Connection with the server is DOWN';
    return (
      <Icon
        icon={icon}
        size="lg"
        title={title}
        style={style.wsStatus(fConnected)}
      />
    );
  }

  // -----------------------------------------------------
  toggleSettings = () => {
    this.setState({ fSettingsShown: !this.state.fSettingsShown });
  }

  onChangeQuickFind = (ev) => {
    this.props.quickFind(ev.target.value);
  }
}

// -----------------------------------------------------
const style = {
  outer: (colors) => {
    const rulerColor = isDark(colors.colorUiBg) ? '#ccc' : '#555';
    return {
      position: 'fixed',
      top: 0,
      left: 0,
      height: 30,
      width: '100%',
      backgroundColor: colors.colorUiBg,
      borderBottom: `1px solid ${rulerColor}`,
      display: 'flex',
      flexDirection: 'row',
      whiteSpace: 'nowrap',
      zIndex: 10,
    };
  },
  icon: (colors) => ({
    cursor: 'pointer',
    color: colors.colorUiFg,
    marginRight: 10,
  }),
  wsStatus: (fConnected) => ({
    marginRight: 5,
    marginLeft: 10,
    color: fConnected ? 'green' : 'red',
    cursor: 'default',
  }),
  placeholder: { height: 30 },
  left: { padding: '4px 4px 4px 8px' },
  right: { padding: '4px 8px 4px 4px' },
  spacer: { flex: '1 1 0px' },
  quickFind: (colors) => ({
    backgroundColor: 'transparent',
    color: colors.colorUiFg,
    borderWidth: 1,
  }),
};

// -----------------------------------------------------
const connect = ReactRedux.connect(mapStateToProps, mapDispatchToProps);
export default connect(Toolbar);
export { Toolbar as _Toolbar };
