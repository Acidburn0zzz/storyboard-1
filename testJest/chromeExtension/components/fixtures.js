import { addDefaults } from 'timm';

export const BASE_COLORS = {
  colorClientBg: 'aliceblue',
  colorClientBgIsDark: false,
  colorClientFg: 'black',
  colorServerBg: 'rgb(214, 236, 255)',
  colorServerBgIsDark: false,
  colorServerFg: 'black',
  colorUiBg: 'white',
  colorUiBgIsDark: false,
  colorUiFg: 'rgb(64, 64, 64)',
};

export const DARK_COLORS = {
  colorClientBg: 'rgb(17, 22, 54)',
  colorClientBgIsDark: true,
  colorClientFg: 'white',
  colorServerBg: 'rgb(14, 11, 33)',
  colorServerBgIsDark: true,
  colorServerFg: 'white',
  colorUiBg: 'black',
  colorUiBgIsDark: true,
  colorUiFg: 'white',
};

export const EMPTY_MAIN_STORY = {
  fExpanded: true,
  fHierarchical: true,
  fOpen: true,
  fWrapper: true,
  records: [
    {
      fExpanded: true,
      fHierarchical: true,
      fMain: true,
      fOpen: true,
      fServer: false,
      fStoryObject: true,
      id: 'main_0',
      lastAction: 'CREATED',
      numRecords: 0,
      pathStr: 'records/0',
      records: [],
      src: 'main',
      status: undefined,
      storyId: 'main_0',
      t: 0,
      title: 'Client',
    },
    {
      fExpanded: true,
      fHierarchical: true,
      fMain: true,
      fOpen: true,
      fServer: true,
      fStoryObject: true,
      id: 'main_1',
      lastAction: 'CREATED',
      numRecords: 0,
      pathStr: 'records/1',
      records: [],
      src: 'main',
      status: undefined,
      storyId: 'main_1',
      t: 0,
      title: 'Server',
    },
  ],
};

export const buildLogRecord = (options = {}) => addDefaults(options, {
  id: 'id1',
  hubId: 'hubId1',
  version: 3,
  fStory: false,
  fServer: false,
  storyId: 'story1',
  t: 0,
  src: 'main',
  level: 30,
  msg: 'Example message',
});

export const buildStory = (options = {}) => {
  const story = addDefaults(options, {
    id: 'id1',
    hubId: 'hubId1',
    version: 3,
    fStory: true,
    fServer: false,
    storyId: 'story1',
    t: 0,
    src: 'main',
    level: 30,
    fRoot: false,
    title: 'Story title',
    parents: [],
    // client-side
    fStoryObject: true,
    fOpen: false,
    fMain: false,
    status: undefined,
    fExpanded: true,
    fHierarchical: true,
    records: [],
  });
  story.records.unshift({
    id: `${story.id}.creation`,
    hubId: story.hubId,
    version: story.version,
    fStory: true,
    fServer: story.fServer,
    storyId: story.storyId,
    t: story.t,
    src: story.src,
    level: story.level,
    fRoot: story.fRoot,
    title: story.title,
    action: 'CREATED',
  });
  if (!story.fOpen) {
    story.records.push({
      id: `${story.id}.closure`,
      hubId: story.hubId,
      version: story.version,
      fStory: true,
      fServer: story.fServer,
      storyId: story.storyId,
      t: story.records[story.records.length - 1].t,
      src: story.src,
      level: story.level,
      fRoot: story.fRoot,
      title: story.title,
      action: 'CLOSED',
    });
  }
  story.numRecords = story.records.length;
  return story;
};
