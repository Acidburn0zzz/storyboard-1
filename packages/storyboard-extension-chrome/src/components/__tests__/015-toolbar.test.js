/* eslint-env jest */
/* eslint-disable global-require, import/newline-after-import */
import React from 'react';
import renderer from 'react-test-renderer';
import { _Toolbar as Toolbar } from '../015-toolbar';
import { BASE_COLORS } from './fixtures';
import $ from './jestQuery';

jest.mock('react-dom');
jest.mock('../010-login', () => require('./mockComponent')('Login'));
jest.mock('../016-settings', () => require('./mockComponent')('Settings'));

describe('Toolbar', () => {
  it('renders correctly when disconnected', () => {
    const tree = renderer.create(
      <Toolbar
        colors={BASE_COLORS}
        wsState="DISCONNECTED"
        expandAllStories={() => {}}
        collapseAllStories={() => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when connected', () => {
    const tree = renderer.create(
      <Toolbar
        colors={BASE_COLORS}
        wsState="CONNECTED"
        expandAllStories={() => {}}
        collapseAllStories={() => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows settings when the corresponding icon is clicked', () => {
    const component = renderer.create(
      <Toolbar
        colors={BASE_COLORS}
        wsState="CONNECTED"
        expandAllStories={() => {}}
        collapseAllStories={() => {}}
      />
    );
    let tree = component.toJSON();
    const showSettingsBtn = $(tree, '#sbBtnShowSettings');
    expect(showSettingsBtn).not.toBeNull();
    showSettingsBtn.props.onClick();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
