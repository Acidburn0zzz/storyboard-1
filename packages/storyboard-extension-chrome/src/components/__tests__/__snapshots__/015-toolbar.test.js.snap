exports[`Toolbar renders correctly when connected 1`] = `
<div
  id="sbToolbar">
  <div
    style={
      Object {
        "backgroundColor": "white",
        "borderBottom": "1px solid #555",
        "display": "flex",
        "flexDirection": "row",
        "height": 30,
        "left": 0,
        "position": "fixed",
        "top": 0,
        "whiteSpace": "nowrap",
        "width": "100%",
        "zIndex": 10,
      }
    }>
    <div
      style={
        Object {
          "padding": "4px 4px 4px 8px",
        }
      }>
      <i
        className="giu-icon fa fa-cog fa-lg"
        id="sbBtnShowSettings"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(64, 64, 64)",
            "cursor": "pointer",
            "letterSpacing": "normal",
            "marginRight": 10,
          }
        }
        title="Show settings..." />
      <i
        className="giu-icon fa fa-chevron-circle-down fa-lg"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(64, 64, 64)",
            "cursor": "pointer",
            "letterSpacing": "normal",
            "marginRight": 10,
          }
        }
        title="Expand all stories" />
      <i
        className="giu-icon fa fa-chevron-circle-right fa-lg"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(64, 64, 64)",
            "cursor": "pointer",
            "letterSpacing": "normal",
            "marginRight": 10,
          }
        }
        title="Collapse all stories" />
      <i
        className="giu-icon fa fa-remove fa-lg"
        onClick={undefined}
        style={
          Object {
            "color": "rgb(64, 64, 64)",
            "cursor": "pointer",
            "letterSpacing": "normal",
            "marginRight": 10,
          }
        }
        title="Clear logs" />
      <i
        className="giu-icon fa fa-info-circle fa-lg"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(64, 64, 64)",
            "cursor": "pointer",
            "letterSpacing": "normal",
            "marginRight": 10,
          }
        }
        title="Show hints" />
       
      <input
        id="quickFind"
        onChange={[Function]}
        placeholder="Quick find..."
        results={0}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 1,
            "color": "rgb(64, 64, 64)",
          }
        }
        type="search" />
      <i
        className="giu-icon fa fa-chain fa-lg"
        id="sbWsStatusIcon"
        style={
          Object {
            "color": "green",
            "cursor": "default",
            "letterSpacing": "normal",
            "marginLeft": 10,
            "marginRight": 5,
          }
        }
        title="Connection with the server is UP" />
    </div>
    <div
      style={
        Object {
          "flex": "1 1 0px",
        }
      } />
    <div
      colors={
        Object {
          "colorClientBg": "aliceblue",
          "colorClientBgIsDark": false,
          "colorClientFg": "black",
          "colorServerBg": "rgb(214, 236, 255)",
          "colorServerBgIsDark": false,
          "colorServerFg": "black",
          "colorUiBg": "white",
          "colorUiBgIsDark": false,
          "colorUiFg": "rgb(64, 64, 64)",
        }
      }
      dataMockType="Login">
      <div
        style={
          Object {
            "color": "blue",
            "fontWeight": "bold",
          }
        }>
        Login
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "height": 30,
      }
    } />
</div>
------------HTML PREVIEW---------------
<div id="sbToolbar">
  <div style="background-color:white;border-bottom:1px solid #555;display:flex;flex-direction:row;height:30px;left:0;position:fixed;top:0;white-space:nowrap;width:100%;z-index:10;">
    <div style="padding:4px 4px 4px 8px;">
      <i
        class="giu-icon fa fa-cog fa-lg"
        id="sbBtnShowSettings"
        style="color:rgb(64, 64, 64);cursor:pointer;letter-spacing:normal;margin-right:10px;"
        title="Show settings..."
      >
      </i>
      <i
        class="giu-icon fa fa-chevron-circle-down fa-lg"
        style="color:rgb(64, 64, 64);cursor:pointer;letter-spacing:normal;margin-right:10px;"
        title="Expand all stories"
      >
      </i>
      <i
        class="giu-icon fa fa-chevron-circle-right fa-lg"
        style="color:rgb(64, 64, 64);cursor:pointer;letter-spacing:normal;margin-right:10px;"
        title="Collapse all stories"
      >
      </i>
      <i
        class="giu-icon fa fa-remove fa-lg"
        style="color:rgb(64, 64, 64);cursor:pointer;letter-spacing:normal;margin-right:10px;"
        title="Clear logs"
      >
      </i>
      <i
        class="giu-icon fa fa-info-circle fa-lg"
        style="color:rgb(64, 64, 64);cursor:pointer;letter-spacing:normal;margin-right:10px;"
        title="Show hints"
      >
      </i>
       
      <input
        id="quickFind"
        placeholder="Quick find..."
        style="background-color:transparent;border-width:1px;color:rgb(64, 64, 64);"
        type="search"
      >
      <i
        class="giu-icon fa fa-chain fa-lg"
        id="sbWsStatusIcon"
        style="color:green;cursor:default;letter-spacing:normal;margin-left:10px;margin-right:5px;"
        title="Connection with the server is UP"
      >
      </i>
    </div>
    <div style="flex:1 1 0px;"></div>
    <div dataMockType="Login">
      <div style="color:blue;font-weight:bold;">
        Login
      </div>
    </div>
  </div>
  <div style="height:30px;"></div>
</div>
`;

exports[`Toolbar renders correctly when disconnected 1`] = `
<div
  id="sbToolbar">
  <div
    style={
      Object {
        "backgroundColor": "white",
        "borderBottom": "1px solid #555",
        "display": "flex",
        "flexDirection": "row",
        "height": 30,
        "left": 0,
        "position": "fixed",
        "top": 0,
        "whiteSpace": "nowrap",
        "width": "100%",
        "zIndex": 10,
      }
    }>
    <div
      style={
        Object {
          "padding": "4px 4px 4px 8px",
        }
      }>
      <i
        className="giu-icon fa fa-cog fa-lg"
        id="sbBtnShowSettings"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(64, 64, 64)",
            "cursor": "pointer",
            "letterSpacing": "normal",
            "marginRight": 10,
          }
        }
        title="Show settings..." />
      <i
        className="giu-icon fa fa-chevron-circle-down fa-lg"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(64, 64, 64)",
            "cursor": "pointer",
            "letterSpacing": "normal",
            "marginRight": 10,
          }
        }
        title="Expand all stories" />
      <i
        className="giu-icon fa fa-chevron-circle-right fa-lg"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(64, 64, 64)",
            "cursor": "pointer",
            "letterSpacing": "normal",
            "marginRight": 10,
          }
        }
        title="Collapse all stories" />
      <i
        className="giu-icon fa fa-remove fa-lg"
        onClick={undefined}
        style={
          Object {
            "color": "rgb(64, 64, 64)",
            "cursor": "pointer",
            "letterSpacing": "normal",
            "marginRight": 10,
          }
        }
        title="Clear logs" />
      <i
        className="giu-icon fa fa-info-circle fa-lg"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(64, 64, 64)",
            "cursor": "pointer",
            "letterSpacing": "normal",
            "marginRight": 10,
          }
        }
        title="Show hints" />
       
      <input
        id="quickFind"
        onChange={[Function]}
        placeholder="Quick find..."
        results={0}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 1,
            "color": "rgb(64, 64, 64)",
          }
        }
        type="search" />
      <i
        className="giu-icon fa fa-chain-broken fa-lg"
        id="sbWsStatusIcon"
        style={
          Object {
            "color": "red",
            "cursor": "default",
            "letterSpacing": "normal",
            "marginLeft": 10,
            "marginRight": 5,
          }
        }
        title="Connection with the server is DOWN" />
    </div>
    <div
      style={
        Object {
          "flex": "1 1 0px",
        }
      } />
    <div
      colors={
        Object {
          "colorClientBg": "aliceblue",
          "colorClientBgIsDark": false,
          "colorClientFg": "black",
          "colorServerBg": "rgb(214, 236, 255)",
          "colorServerBgIsDark": false,
          "colorServerFg": "black",
          "colorUiBg": "white",
          "colorUiBgIsDark": false,
          "colorUiFg": "rgb(64, 64, 64)",
        }
      }
      dataMockType="Login">
      <div
        style={
          Object {
            "color": "blue",
            "fontWeight": "bold",
          }
        }>
        Login
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "height": 30,
      }
    } />
</div>
------------HTML PREVIEW---------------
<div id="sbToolbar">
  <div style="background-color:white;border-bottom:1px solid #555;display:flex;flex-direction:row;height:30px;left:0;position:fixed;top:0;white-space:nowrap;width:100%;z-index:10;">
    <div style="padding:4px 4px 4px 8px;">
      <i
        class="giu-icon fa fa-cog fa-lg"
        id="sbBtnShowSettings"
        style="color:rgb(64, 64, 64);cursor:pointer;letter-spacing:normal;margin-right:10px;"
        title="Show settings..."
      >
      </i>
      <i
        class="giu-icon fa fa-chevron-circle-down fa-lg"
        style="color:rgb(64, 64, 64);cursor:pointer;letter-spacing:normal;margin-right:10px;"
        title="Expand all stories"
      >
      </i>
      <i
        class="giu-icon fa fa-chevron-circle-right fa-lg"
        style="color:rgb(64, 64, 64);cursor:pointer;letter-spacing:normal;margin-right:10px;"
        title="Collapse all stories"
      >
      </i>
      <i
        class="giu-icon fa fa-remove fa-lg"
        style="color:rgb(64, 64, 64);cursor:pointer;letter-spacing:normal;margin-right:10px;"
        title="Clear logs"
      >
      </i>
      <i
        class="giu-icon fa fa-info-circle fa-lg"
        style="color:rgb(64, 64, 64);cursor:pointer;letter-spacing:normal;margin-right:10px;"
        title="Show hints"
      >
      </i>
       
      <input
        id="quickFind"
        placeholder="Quick find..."
        style="background-color:transparent;border-width:1px;color:rgb(64, 64, 64);"
        type="search"
      >
      <i
        class="giu-icon fa fa-chain-broken fa-lg"
        id="sbWsStatusIcon"
        style="color:red;cursor:default;letter-spacing:normal;margin-left:10px;margin-right:5px;"
        title="Connection with the server is DOWN"
      >
      </i>
    </div>
    <div style="flex:1 1 0px;"></div>
    <div dataMockType="Login">
      <div style="color:blue;font-weight:bold;">
        Login
      </div>
    </div>
  </div>
  <div style="height:30px;"></div>
</div>
`;

exports[`Toolbar shows settings when the corresponding icon is clicked 1`] = `
<div
  id="sbToolbar">
  <div
    colors={
      Object {
        "colorClientBg": "aliceblue",
        "colorClientBgIsDark": false,
        "colorClientFg": "black",
        "colorServerBg": "rgb(214, 236, 255)",
        "colorServerBgIsDark": false,
        "colorServerFg": "black",
        "colorUiBg": "white",
        "colorUiBgIsDark": false,
        "colorUiFg": "rgb(64, 64, 64)",
      }
    }
    dataMockType="Settings"
    onClose={[Function]}>
    <div
      style={
        Object {
          "color": "blue",
          "fontWeight": "bold",
        }
      }>
      Settings
    </div>
  </div>
  <div
    style={
      Object {
        "backgroundColor": "white",
        "borderBottom": "1px solid #555",
        "display": "flex",
        "flexDirection": "row",
        "height": 30,
        "left": 0,
        "position": "fixed",
        "top": 0,
        "whiteSpace": "nowrap",
        "width": "100%",
        "zIndex": 10,
      }
    }>
    <div
      style={
        Object {
          "padding": "4px 4px 4px 8px",
        }
      }>
      <i
        className="giu-icon fa fa-cog fa-lg"
        id="sbBtnShowSettings"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(64, 64, 64)",
            "cursor": "pointer",
            "letterSpacing": "normal",
            "marginRight": 10,
          }
        }
        title="Show settings..." />
      <i
        className="giu-icon fa fa-chevron-circle-down fa-lg"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(64, 64, 64)",
            "cursor": "pointer",
            "letterSpacing": "normal",
            "marginRight": 10,
          }
        }
        title="Expand all stories" />
      <i
        className="giu-icon fa fa-chevron-circle-right fa-lg"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(64, 64, 64)",
            "cursor": "pointer",
            "letterSpacing": "normal",
            "marginRight": 10,
          }
        }
        title="Collapse all stories" />
      <i
        className="giu-icon fa fa-remove fa-lg"
        onClick={undefined}
        style={
          Object {
            "color": "rgb(64, 64, 64)",
            "cursor": "pointer",
            "letterSpacing": "normal",
            "marginRight": 10,
          }
        }
        title="Clear logs" />
      <i
        className="giu-icon fa fa-info-circle fa-lg"
        onClick={[Function]}
        style={
          Object {
            "color": "rgb(64, 64, 64)",
            "cursor": "pointer",
            "letterSpacing": "normal",
            "marginRight": 10,
          }
        }
        title="Show hints" />
       
      <input
        id="quickFind"
        onChange={[Function]}
        placeholder="Quick find..."
        results={0}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 1,
            "color": "rgb(64, 64, 64)",
          }
        }
        type="search" />
      <i
        className="giu-icon fa fa-chain fa-lg"
        id="sbWsStatusIcon"
        style={
          Object {
            "color": "green",
            "cursor": "default",
            "letterSpacing": "normal",
            "marginLeft": 10,
            "marginRight": 5,
          }
        }
        title="Connection with the server is UP" />
    </div>
    <div
      style={
        Object {
          "flex": "1 1 0px",
        }
      } />
    <div
      colors={
        Object {
          "colorClientBg": "aliceblue",
          "colorClientBgIsDark": false,
          "colorClientFg": "black",
          "colorServerBg": "rgb(214, 236, 255)",
          "colorServerBgIsDark": false,
          "colorServerFg": "black",
          "colorUiBg": "white",
          "colorUiBgIsDark": false,
          "colorUiFg": "rgb(64, 64, 64)",
        }
      }
      dataMockType="Login">
      <div
        style={
          Object {
            "color": "blue",
            "fontWeight": "bold",
          }
        }>
        Login
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "height": 30,
      }
    } />
</div>
------------HTML PREVIEW---------------
<div id="sbToolbar">
  <div dataMockType="Settings">
    <div style="color:blue;font-weight:bold;">
      Settings
    </div>
  </div>
  <div style="background-color:white;border-bottom:1px solid #555;display:flex;flex-direction:row;height:30px;left:0;position:fixed;top:0;white-space:nowrap;width:100%;z-index:10;">
    <div style="padding:4px 4px 4px 8px;">
      <i
        class="giu-icon fa fa-cog fa-lg"
        id="sbBtnShowSettings"
        style="color:rgb(64, 64, 64);cursor:pointer;letter-spacing:normal;margin-right:10px;"
        title="Show settings..."
      >
      </i>
      <i
        class="giu-icon fa fa-chevron-circle-down fa-lg"
        style="color:rgb(64, 64, 64);cursor:pointer;letter-spacing:normal;margin-right:10px;"
        title="Expand all stories"
      >
      </i>
      <i
        class="giu-icon fa fa-chevron-circle-right fa-lg"
        style="color:rgb(64, 64, 64);cursor:pointer;letter-spacing:normal;margin-right:10px;"
        title="Collapse all stories"
      >
      </i>
      <i
        class="giu-icon fa fa-remove fa-lg"
        style="color:rgb(64, 64, 64);cursor:pointer;letter-spacing:normal;margin-right:10px;"
        title="Clear logs"
      >
      </i>
      <i
        class="giu-icon fa fa-info-circle fa-lg"
        style="color:rgb(64, 64, 64);cursor:pointer;letter-spacing:normal;margin-right:10px;"
        title="Show hints"
      >
      </i>
       
      <input
        id="quickFind"
        placeholder="Quick find..."
        style="background-color:transparent;border-width:1px;color:rgb(64, 64, 64);"
        type="search"
      >
      <i
        class="giu-icon fa fa-chain fa-lg"
        id="sbWsStatusIcon"
        style="color:green;cursor:default;letter-spacing:normal;margin-left:10px;margin-right:5px;"
        title="Connection with the server is UP"
      >
      </i>
    </div>
    <div style="flex:1 1 0px;"></div>
    <div dataMockType="Login">
      <div style="color:blue;font-weight:bold;">
        Login
      </div>
    </div>
  </div>
  <div style="height:30px;"></div>
</div>
`;
